metals/src/main/scala/scala/meta/internal/metals/onTypeRangeFormatters/OnTypeRangeFormatter.scala

```diff
@@ -0,0 +1,36 @@
+package scala.meta.internal.metals.onTypeRangeFormatters
+
+import scala.meta.tokens.Tokens
+
+import org.eclipse.lsp4j.FormattingOptions
+import org.eclipse.lsp4j.Position
+import org.eclipse.lsp4j.Range
+import org.eclipse.lsp4j.TextEdit
+
+trait OnTypeRangeFormatter {
+  type StartPosition = meta.Position
+  type EndPosition = meta.Position
+
+  def onRangeContribute(
+      sourceText: String,
+      range: Range,
+      splitLines: Array[String],
+      startPos: StartPosition,
+      endPos: EndPosition,
+      formattingOptions: FormattingOptions,
+      tokens: Option[Tokens]
+  ): Option[List[TextEdit]] = None
+
+  def onTypeContribute(
+      sourceText: String,
+      range: Range,
+      splitLines: Array[String],
+      startPos: StartPosition,
+      endPos: EndPosition,
+      triggerChar: String,
+      position: Position,
+      enableStripMargin: Boolean,
```

Comment:
```suggestion
```
This is needed for a single class, I would just have it as a fucntion parameter of `MultilineString`

Reply:
I tried to add it as a `MultilineString` parameters, but that means it's gonna be a `case class` and not anymore an `object`, as you suggested.

Reply:
It's fine, it can be the case class then.